service/storage_service.cc

```diff
@@ -1001,21 +1001,33 @@ void storage_service::handle_state_normal(inet_address endpoint) {
             (host_id == _gossiper.get_host_id(db().local().get_replace_address().value()))) {
```

Comment:
This commit should be sent separately, not part of this series. And we need a github issue for it.

Reply:
Do you mean "storage_service: Unify handling of replaced node removal from gossip" should be in a separate PR? Without it, this PR won't work. It is part of the solution to make boot faster and safer.

Reply:
No, I meant "storage_service: Prevent a replaced node rejoins the cluster again  …". 

I forgot that github doesn't associate comments with commits.

Reply:
Sure. "storage_service: Prevent a replaced node rejoins the cluster again" can be in a separate PR. I am removing it from this PR.